### PR TITLE
enable komed custom sound on ios

### PR DESCRIPTION
**Pull request by kmyllyvi**
https://komed-health.atlassian.net/browse/KOM-3748

This modification plays the komed default sound file if it's saved on app  resources (done on Komed Cordova plugin) - if the doesn't exist it will still play the iOS default sound.
Also added support for using "priority" sound - which must be specifically triggered with "priority" string option

### CODE REVIEW

File: src/ios/APPNotificationOptions.m [hunk 1]
```diff
@@ -160,32 +160,21 @@ - (NSString*) actionGroupId
 
 /**
  * The sound file for the notification.
- *
+ * Sound option (string) can be given in options.
  * @return [ UNNotificationSound* ]
  */
 - (UNNotificationSound*) sound
```
Comment (BorntraegerMarc) [inline src/ios/APPNotificationOptions.m:168]: so passing the string `komed_notification_default.wav` didn't work here?

**Comment by BorntraegerMarc** [comment]
> This modification plays the komed default sound file if it's saved on app resources (done on Komed Cordova plugin) - if the doesn't exist it will still play the iOS default sound.
Also added support for using "priority" sound - which must be specifically triggered with "priority" string option

maybe you should comment this in the code. so in the future we remember this

File: src/ios/APPNotificationOptions.m (hunk 1 above)
Comment (kmyllyvi) [reply src/ios/APPNotificationOptions.m:168]: that would work sure, but then we would need to pass it and change the code on FE and komed plugin. This way only one code base is needs to be changed (here). I could make it dynamic also and just use default if fileName is not provided. This was just the fastest way to get the result.

Comment (kmyllyvi) [reply src/ios/APPNotificationOptions.m:168]: but to allow passing filename won't be a biggy so I can do that

Comment (kmyllyvi) [reply src/ios/APPNotificationOptions.m:168]: Okay it's now more dynamic as the method description also says - so it's possible to override the default sound by providing a filename

File: plugin.xml [hunk 2]
```diff
@@ -87,6 +87,9 @@
 
         <header-file src="src/ios/UNNotificationRequest+APPLocalNotification.h" />
         <source-file src="src/ios/UNNotificationRequest+APPLocalNotification.m" />
+
+        <resource-file src="src/sounds/komed_notification_default.wav" target="komed_notification_default.wav"/>
```
Comment (BorntraegerMarc) [inline plugin.xml:91]: so you decided to add these resource files after all in this plugin and not the komed-plugin-push?

**Review by BorntraegerMarc — APPROVED** [review]

**Comment by BorntraegerMarc** [comment]
and did you reach out to the android dev to take care of the android part?

File: plugin.xml (hunk 2 above)
Comment (kmyllyvi) [reply plugin.xml:91]: yes. I think it's better to have the reference to file in the same repo as the file

**Comment by kmyllyvi** [comment]
Android guy pinged - he said he's available

**Pull request merged**